src/box/vy_log.c

```diff
@@ -2393,22 +2402,25 @@ vy_recovery_new_f(va_list ap)
 	struct xrow_header row;
 	while ((rc = xlog_cursor_next(&cursor, &row, false)) == 0) {
 		struct vy_log_record record;
+		size_t region_svp = region_used(&fiber()->gc);
 		rc = vy_log_record_decode(&record, &row);
-		if (rc < 0)
+		if (rc < 0) {
+			region_truncate(&fiber()->gc, region_svp);
 			break;
+		}
 		say_verbose("load vylog record: %s",
 			    vy_log_record_str(&record));
 		if (record.type == VY_LOG_SNAPSHOT) {
+			region_truncate(&fiber()->gc, region_svp);
 			if ((flags & VY_RECOVERY_LOAD_CHECKPOINT) != 0)
 				break;
 			continue;
 		}
 		rc = vy_recovery_process_record(recovery, &record);
+		region_truncate(&fiber()->gc, region_svp);
 		if (rc < 0)
 			break;
-		fiber_gc();
 	}
-	fiber_gc();
```

Comment:
Nit: Why not set `region_svp` outside this loop and simply replace these `fiber_gc` calls with `region_truncate`? This would look less intrusive.

Reply:
Replaced with truncation at the beginning of the loop and after exiting the loop.